firmware/hw_layer/ports/stm32/stm32_common.cpp

```diff
@@ -252,6 +252,11 @@ stm32_hardware_pwm pwmChannels[] = {
 
 /*static*/ hardware_pwm* hardware_pwm::tryInitPin(const char* msg, brain_pin_e pin, float frequencyHz, float duty)
 {
+	// Slow PWM isn't worth doing on hardware - reserve that timer for something faster if needed.
+	if (frequencyHz < 100) {
```

Comment:
that's really weird to be choosing HW at runtime. Does not look right to me.

Reply:
You mean like how we already do this for ICU inputs, EXTI inputs, SPI devices, CAN devices, serial devices, etc?

Reply:
Looks a bit different to me - in ICU inputs, EXTI inputs, SPI devices, CAN devices, serial devices we select HW _directly_ based on _HW_ parameters

here we choose HW based on something not obviously connected to HW.

Reply:
comment & threshold updated to the actual limit of what this implementation can do